tests/capture/test_capture_while_loop.py

```diff
@@ -344,3 +322,167 @@ def f(x):
     out = f(x0)
     assert list(out.keys()) == ["x"]
     assert qml.math.allclose(out["x"], 10)
+
+
+@pytest.mark.usefixtures("enable_disable_dynamic_shapes")
+class TestCaptureWhileLoopDynamicShapes:
+
+    def test_while_loop_dynamic_shape_array(self):
+        """Test while loop can accept arrays with dynamic shapes."""
+
+        def f(x):
+            @qml.while_loop(lambda res: jax.numpy.sum(res) < 10)
```

Comment:
From the import up top :)
```suggestion
            @qml.while_loop(lambda res: jnp.sum(res) < 10)
```